### PR TITLE
chore: Fill in the NodeClassReference for all NodePools and NodeClaims

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

This updates our NodeClassReferences to include all fields for APIVersion and Kind

**How was this change tested?**

`make presubmit`
`/karpenter snapshot`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: test/pkg/environment/common/environment.go [hunk 1]
```diff
@@ -133,8 +134,11 @@ func NewClient(ctx context.Context, config *rest.Config) client.Client {
 
 func (env *Environment) DefaultNodePool(nodeClass *v1beta1.EC2NodeClass) *corev1beta1.NodePool {
 	nodePool := coretest.NodePool()
+	gvk := object.GVK(nodeClass)
 	nodePool.Spec.Template.Spec.NodeClassRef = &corev1beta1.NodeClassReference{
-		Name: nodeClass.Name,
+		APIVersion: gvk.GroupVersion().String(),
+		Kind:       gvk.Kind,
```
Comment (jigisha620) [inline test/pkg/environment/common/environment.go:140]: Inline this just like you suggested in another PR?

File: test/suites/scale/deprovisioning_test.go [hunk 2]
```diff
@@ -196,8 +197,11 @@ var _ = Describe("Deprovisioning", Label(debug.NoWatch), Label(debug.NoEvents),
 			// the other nodePools
 			driftNodeClass := awstest.EC2NodeClass()
 			driftNodeClass.Spec = *nodeClass.Spec.DeepCopy()
+			gvk := object.GVK(nodeClass)
 			nodePoolMap[driftValue].Spec.Template.Spec.NodeClassRef = &corev1beta1.NodeClassReference{
-				Name: driftNodeClass.Name,
+				APIVersion: gvk.GroupVersion().String(),
+				Kind:       gvk.Kind,
```
Comment (jigisha620) [inline test/suites/scale/deprovisioning_test.go:203]: Inline this too?

File: test/suites/scale/deprovisioning_test.go [hunk 3]
```diff
@@ -196,8 +197,11 @@ var _ = Describe("Deprovisioning", Label(debug.NoWatch), Label(debug.NoEvents),
 			// the other nodePools
 			driftNodeClass := awstest.EC2NodeClass()
 			driftNodeClass.Spec = *nodeClass.Spec.DeepCopy()
+			gvk := object.GVK(nodeClass)
```
Comment (jigisha620) [inline test/suites/scale/deprovisioning_test.go:200]: Should we check GVK for `driftNodeClass` here?

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jigisha620 — APPROVED** [review]
LGTM 🚀

**Pull request merged**